lib/streamlit/runtime/state/session_state.py

```diff
@@ -606,6 +608,14 @@ def get_stats(self) -> list[CacheStat]:
         stat = CacheStat("st_session_state", "", asizeof(self))
         return [stat]
 
+    def _check_serializable(self) -> None:
```

Comment:
Can you add a short docstring to this (and `maybe_check_serializable`)? I think we want to call out:

- When and why we check for serializability
- How we check for serializability (pickle)
- What happens when the check fails (we throw an exception - also, let's call out where we expect this exception to be caught and displayed)